backend/src/tasks/LookingGlass.ts

```diff
@@ -0,0 +1,286 @@
+
+  const collections: LGCollectionsResponse[] = await collectionByWorkspace();
+  const ipsAndDomains: string[] = [];
+  const domains: Domain[] = [];
+  const Vulns_list: any = [];
+  const Vulnerabilities: Vulnerability[] = [];
+
+  for (const line of collections) {
+    //Match the organization to the LookingGlass Collection
+    if (organizationName == line['name']) {
+      const collectionID = line['id'];
+      console.log(line['name']);
+      //Query LookingGlass for the Threat info
+      const data: LGThreatResponse[] = await getThreatInfo(collectionID);
+      const responseDomains: Domain[] = await saveAndPullDomains(
+        data,
+        organizationId,
+        scanId,
+        Org
+      );
+      for (const l of data['results']) {
+        //Create a dictionary of relevant fields from the API request
+        if (typeof Org === 'object') {
+          const val = {
+            firstSeen: new Date(l['firstSeen']),
+            lastSeen: new Date(l['lastSeen']),
+            sources: l['sources'],
+            ref_type: l['ref']['type'],
+            ref_right_type: l['ref']['right']['type'],
+            ref_right_id: l['ref']['right']['id'],
+            ref_left_type: l['ref']['left']['type'],
+            ref_left_id: l['ref']['right']['id'],
+
+            right_ticScore: l['right']['ticScore'],
+            right_classifications: l['right']['classifications'],
+            right_name: l['right']['name'],
+
+            left_type: l['left']['type'],
+            left_ticScore: l['left']['ticScore'],
+            left_name: l['left']['name']
+          };
+          if (l['right']['classifications'][0] == 'Vulnerable Service') {
+            val['vulnOrMal'] = 'Vulnerability';
+          } else {
+            val['vulnOrMal'] = 'Malware';
+          }
+
+          //if the Domain already exists and add this vuln to the associated vulnerability
+          if (ipsAndDomains.includes(l['left']['name'])) {
+            for (const Vuln of Vulns_list) {
+              if (l['left']['name'] == Vuln['domain'].name) {
+                Vuln['structuredData']['lookingGlassData'].push(val);
+              }
+            }
+          }
+          //if the Domain hasn't been used create a new Domain and a new Vulnerability
+          else {
+            for (const x of responseDomains) {
+              if (x.name == l['left']['name']) {
+                const current_Domain = x;
+                const V = {
+                  domain: current_Domain,
+                  lastSeen: new Date(Date.now()),
+                  title: 'Looking Glass Data',
+                  state: 'open',
+                  source: 'lookingGlass',
+                  needsPopulation: false,
+                  structuredData: {
+                    lookingGlassData: [val]
+                  },
+                  description: `These are Vulnerabilities and Malware found by LookingGlass for ${organizationName}`
```

Comment:
Let's remove organizationName from the description, as this vulnerability isn't unique to the organization (it's unique to the domain)

```suggestion
                  description: `Vulnerabilities and malware found by LookingGlass.`
```

Reply:
The reason I associated it with the org is because the Scan can only get this info if the org matches a LookingGlass Collection in our account. But I guess it isn't necessary since it is already linked to the Domain that is associated with the Org

Reply:
Yeah, it also feels redundant to add the organization name to the vulnerability when we already know what org the related domain is in.